lib/fizzy/parser.cpp

```diff
@@ -126,6 +126,18 @@ inline parser_result<GlobalType> parse_global_type(const uint8_t* pos, const uin
     return {type, pos};
 }
 
+template <typename T>
+inline parser_result<T> read_const(const uint8_t* pos, const uint8_t* end)
```

Comment:
Is this `read_const` or just `read`/`read_value`/`read_type`?

Reply:
I'm considering `parse_float_value`/`parse_float_const` because it returns `parse_result` and is similar to other parse_* functions above and below.

Reply:
Reading `parse_float_const<uint32_t>(..)` is extremely confusing, so I would suggest to go with `parse_value<uint32_t>` or `parse_type<uint32_t>`, especially as this should do byteswapping should we support big endian targets. Right?

Reply:
If `parse_float_const<float>` and `parse_float_const<double>` works that is fine too.

Reply:
> If `parse_float_const<float>` and `parse_float_const<double>` works that is fine too.

Not without a cast on a calling side.

Reply:
That would work (but I would need to check assembly of that to be sure). But might be confusing as you can/will implement `fN.const` as `iN.const`: https://github.com/wasmx/fizzy/blob/master/lib/fizzy/execute.cpp#L1004-L1005.

Reply:
Renamed to `parse_value`